netmaster/docknet/docknet.go

```diff
@@ -287,6 +287,30 @@ func DeleteDockNetState(tenantName, networkName, serviceName string) error {
 	return dnetOper.Clear()
 }
 
+// GetDocknetState gets the docknet entry from state store
+func GetDocknetState(tenantName, networkName, serviceName string) (*DnetOperState, error) {
+	log.Infof("GetDocknetState for %s.%s.%s", tenantName, networkName, serviceName)
+
+	// Get the state driver
+	stateDriver, err := utils.GetStateDriver()
+	if err != nil {
+		log.Warnf("Couldn't get state driver for docknet %v", err)
+		return nil, err
+	}
+
+	// save docknet oper state
+	dnetOper := DnetOperState{}
+	dnetOper.ID = fmt.Sprintf("%s.%s.%s", tenantName, networkName, serviceName)
+	dnetOper.StateDriver = stateDriver
+
+	// Read the dnet oper state
+	err = dnetOper.Read(dnetOper.ID)
```

Comment:
Is this the correct way or walk thru dnet for matchin tenant/network/group ?
Also it is better to check "key not found" error here

Reply:
Since we know tenant, network and group we can form the key and fetch that particular entry, instead of fetching all the entries and matching them.
The caller of the function is checking for 'key not found' in https://github.com/gkvijay/netplugin/blob/b6a3eb335de81f2b9e94f310735ee15f8d716963/netmaster/objApi/apiController.go#L995